packages/core/src/blockchain/storage-layer.ts

```diff
@@ -249,10 +322,13 @@ export class StorageLayer implements StorageLayerProvider {
       const next = await this.findNextKey(prefix, startKey, undefined)
       if (!next) break
       startKey = next
-      if ((await this.get(next, false)) === StorageValueKind.Deleted) continue
       keys.push(next)
     }
-    return keys
+
+    // value could be deleted on #parent and we have to exclude that. We have to load the values to check it
+    const values = await this.getMany(keys, false);
```

Comment:
Although this may look like optimization, it may lead to failure if RPC limit is hit. A proper different  approach is needed to optimize this part. Best to not include this change to avoid RPC limit issue

Reply:
I couldn't find any RPC limit regarding the amount of keys that a queryStorage can handle. AFAIK RPC limit applies when you send too many requests too quickly.

But this is doing the opposite: It's reducing the amount of requests being sent by chopsticks by bundling all if them into a single `state_queryStorageAt`.

But I had another problem, I'll describe in a separate comment.